go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -475,45 +474,52 @@ func importManagementPortAnnotation(node *kapi.Node) (string, error) {
 	return sriovnet.GetVfRepresentorDPU(fmt.Sprintf("%d", pfId), fmt.Sprintf("%d", vfId))
 }
 
-func createNodeManagementPorts(node *kapi.Node, nodeAnnotator kube.Annotator, waiter *startupWaiter,
-	subnets []*net.IPNet) ([]managementPortEntry, *managementPortConfig, error) {
-	// If netdevice name is not provided in the full mode then management port backed by OVS internal port.
-	// If it is provided then it is backed by VF or SF and need to determine its representor name to plug
-	// into OVS integrational bridge
+// If netdevice name is not provided in the full mode then management port backed 
+// by OVS internal port. If it is provided then it is backed by VF or SF and need
+// to determine its representor name to plug into OVS integrational bridge
+// In case ovn-k is running in DPU mode, it will read the annotation from the host
+// side which should have been exported by ovn-k running in DPU host mode.
+func getMagementPortAndRepresentorName(node *kapi.Node) (string, string, error) {
 	if config.OvnKubeNode.Mode == types.NodeModeFull && config.OvnKubeNode.MgmtPortNetdev != "" {
-		deviceID, err := util.GetDeviceIDFromNetdevice(config.OvnKubeNode.MgmtPortNetdev)
+		netdevName := config.OvnKubeNode.MgmtPortNetdev
+		deviceID, err := util.GetDeviceIDFromNetdevice(netdevName)
 		if err != nil {
 			// Device might had been already renamed to types.K8sMgmtIntfName
-			config.OvnKubeNode.MgmtPortNetdev = types.K8sMgmtIntfName
-			if deviceID, err = util.GetDeviceIDFromNetdevice(config.OvnKubeNode.MgmtPortNetdev); err != nil {
-				return nil, nil, fmt.Errorf("failed to get device id for %s or %s: %v",
+			netdevName = types.K8sMgmtIntfName
+			if deviceID, err = util.GetDeviceIDFromNetdevice(netdevName); err != nil {
+				return "", "", fmt.Errorf("failed to get device id for %s or %s: %v",
 					config.OvnKubeNode.MgmtPortNetdev, types.K8sMgmtIntfName, err)
 			}
 		}
 		rep, err := util.GetFunctionRepresentorName(deviceID)
 		if err != nil {
-			return nil, nil, err
+			return "", "", err
+		}
+		return netdevName, rep, err
+	} else if config.OvnKubeNode.Mode == types.NodeModeDPU {
+		rep, err := importManagementPortAnnotation(node)
+		if err != nil {
+			return "", "", err
 		}
-		config.OvnKubeNode.MgmtPortRepresentor = rep
+		return "", rep, err
+	}
+	return config.OvnKubeNode.MgmtPortNetdev, "", nil
+}
+
+func createNodeManagementPorts(node *kapi.Node, nodeAnnotator kube.Annotator, waiter *startupWaiter,
+	subnets []*net.IPNet) ([]managementPortEntry, *managementPortConfig, error) {
+	netdevName, rep, err := getMagementPortAndRepresentorName(node)
```

Comment:
Spelling `getMagementPortAndRepresentorName` => `getManagementPortAndRepresentorName`
Or we can shorten it `getMgmtPortAndRepName`